cms/topics/utils.py

```diff
@@ -26,3 +54,256 @@ def format_time_series_as_document_list(time_series: StreamValue) -> list[dict[s
         time_series_documents.append(time_series_document)
 
     return time_series_documents
+
+
+class ArticleProcessor:
```

Comment:
It would be nice to get the devs' opinions on if it's worth putting these processor service classes in their own file?

Reply:
https://github.com/ONSdigital/dis-wagtail/pull/366#issuecomment-3349251776

Reply:
Thanks, Mebin, I like the idea of a new file too. I was reading this django style guide: https://github.com/HackSoftware/Django-Styleguide?tab=readme-ov-file#services

It recommends putting our service classes in a dedicated `service.py` under whatever app we have, or if we have many services for an app, then under a service directory.

But then the style guide seems very domain driven design (DDD) oriented.

Or should I just go with a file name:
- `related_article_methodology_processors.py` or something

Reply:
The service and selector pattern can be useful. We could go with:

```
  selectors/
    __init__.py
    related_articles.py    # query-only selectors for articles
    related_methodologies.py   # query-only selectors for methodology
  services/
    __init__.py
    types.py   # for your added types like TypedDict
    related_content.py     # orchestration that calls selectors and shapes output
```
Rename the classes to: `RelatedArticlesService` and `RelatedMethodologiesService`. or "Processor/Resolve" (im impartial).

We don't follow this pattern currently. So the selectors can live with `services/releated_content.py`.

Reply:
As we discussed on Slack, we both like the idea of following the service selector design pattern. As mentioned earlier we don't follow the style guide above/ it's design patterns. It would be good to as we move forwards as it will actually make the project more navigatable and maintainable in the long run.

For now I went with `services/related_content.py` and keep the selectors within the same file too and I have a `types.py` file too.